File: MomentsJava/src/main/java/com/example/momentsjava/data/api/di/ListApiModule.java
package com.example.momentsjava.data.api.di;

import com.example.momentsjava.data.api.ListApiService;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;
import hu.akarnokd.rxjava3.retrofit.RxJava3CallAdapterFactory;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

@Module
@InstallIn(SingletonComponent.class)
public class ListApiModule {
    @Provides
    public static Retrofit provideRetrofit() {
        return new Retrofit.Builder()
                .baseUrl("https://demo-list-server.vercel.app/api/")
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
                .build();
    }

    @Provides
    public static ListApiService provideListApiService(Retrofit retrofit) {
        return retrofit.create(ListApiService.class);
    }
}

File: app/src/main/java/com/example/simplelist/data/api/di/ListApiModule.kt
package com.example.simplelist.data.api.di


import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

@Module
@InstallIn(SingletonComponent::class)
object listApiModule {

    @Provides
    fun provideRetrofit(): Retrofit {
        return Retrofit.Builder()
            .baseUrl("http://10.0.2.2:3022")
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }

    @Provides
    fun provideListApiService(retrofit: Retrofit): ListApiService {
        return retrofit.create(ListApiService::class.java)
    }
}
